### PR TITLE
Update Arithmetic.hpp to fix bug in return units of cross function

### LINKED ISSUE #79: Wrong units generated with Cross product
When `cartesian_representation` is crossed with other types of representation wrong units are generated due to resultant units are generated from original units in the return statement.

### PR DESCRIPTION
**Pull request by nitink25**
### Description

Previously, wrong units were generated for cross function for any combination apart from both Cartesian. That bug has been fixed.

1. A separate function overload has been made to deal with cases where `cartesian` is not first argument.
2. Few mistakes in the tests for `spherical` and `spherical_equatorial` have been fixed too. 

closes #79 .

### CODE REVIEW

File: include/boost/astronomy/coordinate/io.hpp [hunk 1]
```diff
@@ -24,7 +24,7 @@ template
 >
 std::ostream& operator<< (std::ostream &out, cartesian_representation< CoordinateType, XQuantity, YQuantity, ZQuantity> const& point)
 {
-    out << "Cartesian Representation ( " << point.get_x() << " , " << point.get_y() << " , " << point.get_x() << " )";
+    out << "Cartesian Representation ( " << point.get_x() << " , " << point.get_y() << " , " << point.get_z() << " )";
```
Comment (lpranam) [inline include/boost/astronomy/coordinate/io.hpp]: Please create a separate PR for this. It's always good to have unrelated changes submitted differently.

**Review by lpranam — CHANGES_REQUESTED** [review]

File: include/boost/astronomy/coordinate/io.hpp (hunk 1 above)
Comment (nitink25) [reply include/boost/astronomy/coordinate/io.hpp]: Alright... I have removed it from this PR. I will create a new one for it.

File: include/boost/astronomy/coordinate/arithmetic.hpp [hunk 2]
```diff
@@ -120,6 +100,85 @@ auto cross
...
+        typename representation2_type::quantity1::unit_type()))));
+
+    bg::set<2>(result, (bg::get<0>(tempPoint1)*bg::get<1>(tempPoint2)) -
+        ((bg::get<1>(tempPoint1)*
+        bu::conversion_factor(typename representation1_type::quantity2::unit_type(),
+        typename representation1_type::quantity1::unit_type()))*
+        (bg::get<0>(tempPoint2)*
+        bu::conversion_factor(typename representation2_type::quantity1::unit_type(),
+        typename representation2_type::quantity2::unit_type()))));
+
```
Comment (rjnrohit) [inline include/boost/astronomy/coordinate/arithmetic.hpp:115]: I think code snippet from line 138 to line 160 should be changed. Here conversion are not matching with the quantity returning in the return statement.
